lucene/expressions/src/java/org/apache/lucene/expressions/ExpressionFunctionValues.java

```diff
@@ -38,14 +41,23 @@
 
   @Override
   public boolean advanceExact(int doc) throws IOException {
+    if (currentDoc == doc) {
+      return true;
+    }
     for (DoubleValues v : functionValues) {
       v.advanceExact(doc);
     }
+    currentDoc = doc;
+    computed = false;
     return true;
   }
   
   @Override
   public double doubleValue() {
-    return expression.evaluate(functionValues);
+    if (!computed) {
```

Comment:
Can we use `computed == false` instead to improve readability by possibly aged eyes and reduce risk of future bugs?